packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -607,6 +611,63 @@ export const analyzeCollectionSchema = (): CollectionThunkAction<
   };
 };
 
+const validateFakerSchema = async (
+  fakerSchema: MockDataSchemaResponse,
+  logger: Logger
+) => {
+  const { faker } = await import('@faker-js/faker');
```

Comment:
Also can we limit this to the only locale we're using to reduce the size of introduced code?

```suggestion
  const { faker } = await import('@faker-js/faker/locale/en');
```